app/models/concerns/has_recent_items.rb

```diff
@@ -13,8 +13,30 @@ def recent_item_types
       self.class.recent_item_types
     end
 
+    def viewable_map
+      self.class.viewable_map
+    end
+
     def recent_items
-      sorted_recent_item_links.preload(:item).map(&:item).compact
+      recent_item_links.order(updated_at: :desc).preload(:item).map(&:item).compact
+    end
+
+    def viewable_recent_items
+      viewable_recent_item_links.order(updated_at: :desc).preload(:item).map(&:item).compact
+    end
+
+    def viewable_recent_item_links
+      return recent_item_links if viewable_map.compact.empty?
+
+      ids = []
+
+      recent_item_types.values.map do |item_class|
+        ri_scope = recent_item_links.where(item_type: item_class.name)
+        ri_scope = ri_scope.where(item_id: viewable_map[item_class.name].call(self, ri_scope).pluck(:id)) if viewable_map[item_class.name].present?
+        ids += ri_scope.pluck(:id)
+      end
+
+      recent_item_links.where(id: ids)
     end
```

Comment:
I think it's fairly safe to assume that recent items are always going to have some viewability restrictions that are expressed with a `viewable_by` scope. If we assume that, we don't need the `viewable_proc`s at all, right? And it is easier to read:
```
ids += recent_item_links.where(item_type: item_class.name, item_id: item_class.viewable_by(self).pluck(:id))
```

That feels more maintainable to me since the code will be easier to understand. What do you think @ZachCMP @eanders ?

Reply:
It's worth mentioning that the call in the consumer looks like this:

```ruby
has_recent :clients, Hmis::Hud::Client, viewable_proc: ->(user, _ri_scope) { Hmis::Hud::Client.viewable_by(user) }
```

I think that's not so bad to read and gives us the ability to tweak the behavior however we want. However, if there's no use case for that, then I agree that we might as well just use your suggestion. Let me know what you'd like to do.

Reply:
It's worth mentioning that the call in the consumer looks like this:

```ruby
has_recent :clients, Hmis::Hud::Client, viewable_proc: ->(user, _ri_scope) { Hmis::Hud::Client.viewable_by(user) }
```

I think that's not so bad to read and gives us the ability to tweak the behavior however we want. However, if there's no use case for that, then I agree that we might as well just use your suggestion. Let me know what you'd like to do.

Reply:
I guess my inclination is to lean on `viewable_by` for now until we need something else.

Reply:
Implemented Gig's suggestion in the first comment